### PR TITLE
Change .js MIME type to text/javascript

### LINKED ISSUE #34323: Change MIME type for .js extension to text/javascript
According to https://datatracker.ietf.org/doc/html/draft-ietf-dispatch-javascript-mjs which will replace RFC4329, `application/javascript` is obsoleted and `text/javascript` is the recommended MIME type for the `.js` file extension.

Given that this could break clients, I'd recommend against making this change until the draft is ratified. This will likely happen in the .NET 7 timeframe.

👀 @MylesBorins

### PR DESCRIPTION
**Pull request by adityamandaleeka**
Fix #34323

This changes the MIME type for .js to `text/javascript`, and updates response compression to include `text/javascript` as well.

I also added .mjs to the copy of FileExtensionContentTypeProvider.cs that's in WebView sources, which seems to have been missed in #34318.

See:
https://datatracker.ietf.org/doc/html/draft-ietf-dispatch-javascript-mjs
https://html.spec.whatwg.org/multipage/scripting.html#scriptingLanguages

### CODE REVIEW

**Review by javiercn — APPROVED** [review]

**Review by Tratcher — APPROVED** [review]

File: src/Components/WebView/WebView/src/FileExtensionContentTypeProvider.cs
Comment (Eilon) [inline src/Components/WebView/WebView/src/FileExtensionContentTypeProvider.cs:155]: @dotnet/dotnet-maui-blazor-eng - we need to verify this works in all the WebViews. I'd be very surprised if it _doesn't_, but we need to positively verify that it does.

Comment (Eilon) [reply src/Components/WebView/WebView/src/FileExtensionContentTypeProvider.cs:155]: In particular, things like older Androids with older Chromes, which we do still support.

Comment (adityamandaleeka) [reply src/Components/WebView/WebView/src/FileExtensionContentTypeProvider.cs:155]: @Eilon Makes sense. I'm not familiar with your test infrastructure for that stuff; would someone on your team be able to do that test pass before this PR is merged, or would this be picked up post-merge? I'm happy to hold off on merging this if needed, just let me know how to proceed :).

Comment (Eilon) [reply src/Components/WebView/WebView/src/FileExtensionContentTypeProvider.cs:155]: Pinging @TanayParikh @javiercn @MackinnonBuck who can help you with this.

**Pull request merged**
